designs/NuGet-ClientCertificates.md

```diff
@@ -0,0 +1,324 @@
+Can be configured with:
+    - Attribute `packageSource`. Required. `packageSources` source name reference.
+    - Attribute `storeLocation`. [Possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storelocation?view=netframework-4.8#fields). Optional. Equals `CurrentUser` by default.
+    - Attribute `storeName`. [Possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.storename?view=netframework-4.8#fields). Optional. Equals `My` by default.
+    - Attribute `findBy`. [Possible values](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509findtype?view=netframework-4.8#fields) with out FindBy prefix. Optional. Equals `thumbprint` by default.
+    - Attribute `findValue`. `value` - the search criteria as an object. Required.
+
+2. `fileCert` - certificate import from file (DER encoded x.509, Base-64 encoded x.509, PKCS)
+Can be configured with:
+    - Attribute `packageSource`. Required. `packageSources` source name reference.
+    - Attribute `path`. Absolute or relative path to certificate file. Required.
+    - Attribute `password`. Plain or encrypted password string. Optional. Encrypted in same manner as [PackageSourceCredential](https://docs.microsoft.com/en-us/nuget/reference/nuget-config-file#packagesourcecredentials) password.
+
+## Configuration example
+
+```xml
+<configuration>
+    ...
+    <packageSources>
+        <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+        <add key="Contoso" value="https://contoso.com/packages/" />
+        <add key="Foo" value="https://foo.com/bar/packages/" />
+    </packageSources>
+    ...
+    <clientCertificates>	
+        <storeCert packageSource="Contoso"
+                   storeLocation="currentUser"
+                   storeName="my"
+                   findBy="thumbprint" 
+                   findValue="4894671ae5aa84840cc1079e89e82d426bc24ec6" />
+        <fileCert packageSource="Foo"
+                  path=".\certificate.pfx" 
+                  password="..." />
+    </clientCertificates>
+...
+</configuration>
+```
+# NuGet.exe
+
+## client-certificates command
+
+Gets, updates, sets or lists client certificates to the NuGet configuration.
+
+Usage
+```
+nuget client-certificates <list|add|remove|update> [options]
+```
+
+if none of `list|add|remove|update` is specified, the command will default to `list`.
+
+### nuget client-certificates list [options]
```

Comment:
I apologise for not noticing this earlier, but all the filtering options seem more complex than the benefit they provide. Most customers don't have so many sources that filtering is necessary. `nuget sources list`, and `dotnet list packages` don't have filtering options, so adding filtering to this new command feels inconsistent.

Reply:
So are you sure you want to delete filter functionality at all?

Reply:
Done